posts/ultra-portable-arm64/index.md

```diff
@@ -0,0 +1,185 @@
+---
+title: An ultra-portable ARM64 dev environment
+date: May 19, 2023
+author: Marc Paquette
+---
+
+# An ARM64 in every pocket
+
+Raspberry Pi rebooted an important segment of computing for educators and hobbyists. It makes digital technology interactive and fun to use again, and at a reasonable cost for anyone who’s curious and creative.
+
+I’d like to start an education-related project based on ARM64 assembly language, and that’s a problem.
+
+## My problem: Raspberry Pi is in short supply
+
+The pandemic and its supply chain problems conspired to raise demand for the already-popular Raspberry Pi while lowering supply. Things are getting better but finding a Raspberry Pi at a reasonable price isn't easy.
+
+A smaller challenge is carrying a Raspberry Pi around. Setting it up at home is easy. But traveling with it is kind of a pain. To set it up and take it down at a cafe or a hotel room means I have to carry an extra set of cables and a power adapter.
+
+## Solution: An anywhere ARM64 dev environment
+
+What I’d like is an ultra-portable ARM64 device that I can use for development. By “portable” I mean that’s it’s available anywhere. The “ultra” part means that I don’t want to have to carry the physical device and its accessories with me. And I’d like the full desktop experience because my project is all about interactivity.
+
+I chose to solve this with Coder and [AWS EC2 Graviton](https://aws.amazon.com/ec2/graviton/).
+
+- With Coder, I can create a workspace that spins up a container for ARM64 Ubuntu letting me work on my project wherever I want.
+- Graviton is the AWS ARM64-based EC2 service. I’ll run Ubuntu with a desktop environment and access it remotely with VNC.
+- Bonus points for remotely accessing my ultra-portable ARM64 device with just a browser-based VNC client.
+
+This will be an excellent solution while I wait to finally get my hands on a real Raspberry Pi.
+
+## Reuse a wheel instead of reinventing it
+
+A Coder workspace provides a development environment. A Coder template describes what’s in the workspace, how to spin it up, and how to take it down.
```

Comment:
Another value prop we may want to mention is the one-click into VS Code. No need to fiddle around with extensions, ssh keys, or configuring your IDE.

Reply:
I'll add this to step 4.